sycl/include/CL/sycl/detail/queue_impl.hpp

```diff
@@ -141,25 +141,17 @@ class queue_impl {
     RT::PiQueue Queue;
     RT::PiContext Context = detail::getSyclObjImpl(m_Context)->getHandleRef();
     RT::PiDevice Device = detail::getSyclObjImpl(m_Device)->getHandleRef();
-#ifdef CL_VERSION_2_0
-    cl_queue_properties CreationFlagProperties[] = {
-        CL_QUEUE_PROPERTIES, CreationFlags, 0};
-    PI_CALL((Queue = RT::piQueueCreate(
-        Context, Device, CreationFlagProperties, &Error), Error));
-#else
-    // TODO: do we really need this old interface into PI and here?
-    Queue = clCreateCommandQueue(Context, m_Device.get(), CreationFlags, &Error);
-    // TODO catch an exception and put it to list of asynchronous exceptions
-    PI_CHECK(Error);
-#endif
-    // Tf creating out-of-order queue failed and this property is not
+    Error = PI_CALL_RESULT(RT::piQueueCreate(Context, Device, CreationFlags,
+                                             pi::pi_cast<pi_queue *>(&Queue)));
```

Comment:
Not sure if I'm abusing the cast system here, it's possible to just pass it as &Queue, but borrowing a similar style from another PI call.

Reply:
the pi_cast is not needed here